### PR TITLE
[Android] Storage.list must return an immediate error if no network

### PR DESCRIPTION
**Pull request by nikitag-ms**
Please have a look at our [guidelines for contributions](https://github.com/Microsoft/AppCenter-SDK-Android/blob/develop/CONTRIBUTING.md) and consider the following before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.

## Description

Storage.list must return an immediate error if no network

### CODE REVIEW

File: sdk/appcenter-data/src/test/java/com/microsoft/appcenter/data/DataTest.java [hunk 1]
```diff
@@ -151,6 +151,21 @@ public void disablePersisted() {
...
+        /* Make the call. */
+        PaginatedDocuments<TestDocument> docs = Data.list(USER_DOCUMENTS, TestDocument.class).get();
+
+        /* Verify the result correct. */
+        assertFalse(docs.hasNextPage());
+        Page<TestDocument> page = docs.getCurrentPage();
+        assertNull(page.getItems());
+        assertNotNull(page.getError());
+        assertEquals(DataException.class, page.getError().getClass());
+    }
```
Comment (MatkovIvan) [inline sdk/appcenter-data/src/test/java/com/microsoft/appcenter/data/DataTest.java]: In addition, you can add verification that it doesn't try to make HTTP calls

Comment (nikitag-ms) [reply sdk/appcenter-data/src/test/java/com/microsoft/appcenter/data/DataTest.java]: Good point!

File: sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/Data.java [hunk 2]
```diff
@@ -556,6 +556,10 @@ public void onCallFailed(Exception e) {
         if (isInvalidPartitionWhenDocuments(partition, result)) {
             return result;
         }
+        if (!mNetworkStateHelper.isNetworkConnected()) {
+            completeFutureAndRemovePendingCallWhenDocuments(new DataException("No network detected. List operation is not supported offline."), result);
```
Comment (YiMysty) [inline sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/Data.java]: should we complete here? I thought the result is resolved in upper level?

Comment (nikitag-ms) [reply sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/Data.java]: My understanding is that when you want the async operation to return, you complete it. We know we are not going to do anything else and we have the document instance we are going to return to the user. Why not completing it?

**Pull request merged**